sdks/java/core/src/main/java/org/apache/beam/sdk/util/BigQueryTableInserter.java

```diff
@@ -95,6 +94,22 @@ public BigQueryTableInserter(Bigquery client) {
     this.client = client;
     this.defaultRef = null;
     this.maxRowsPerBatch = MAX_ROWS_PER_BATCH;
+    this.executor = new GcsOptions.ExecutorServiceFactory().create(
+            PipelineOptionsFactory.create());
+  }
+
+
+  /**
+   * Constructs a new row inserter.
+   *
+   * @param client a BigQuery client
+   * @param options a PipelineOptions object
+   */
+  public BigQueryTableInserter(Bigquery client, PipelineOptions options) {
+    this.client = client;
+    this.defaultRef = null;
+    this.maxRowsPerBatch = MAX_ROWS_PER_BATCH;
+    this.executor = new GcsOptions.ExecutorServiceFactory().create(options);
```

Comment:
Why not just call:
options.as(GcsOptions.class).getExecutorService() then we will share the same executor instance